daemon/containerd/image_snapshot.go

```diff
@@ -44,6 +47,39 @@ func (i *ImageService) CreateLayerFromImage(img *image.Image, layerName string,
 	return i.createLayer(descriptor, layerName, rwLayerOpts, nil)
 }
 
+func (i *ImageService) generatePrepareOpts(ctx context.Context, rwLayerOpts *layer.CreateRWLayerOpts) ([]snapshots.Opt, error) {
+	var opts []snapshots.Opt
+
+	if rwLayerOpts != nil && len(rwLayerOpts.StorageOpt) > 0 {
+		for key, val := range rwLayerOpts.StorageOpt {
+			key = strings.ToLower(key)
+			switch key {
+			case "size":
```

Comment:
We should consider making these strict and not case-sensitive (but it's possible that the old code was also relaxed); I don't think we documented that these were handled case-insensitive, so perhaps it's fine to (for the containerd snapshotter) just being strict.

Reply:
I used the overlay2 quota behavior as a guide, but I don't have any strong opinion here on case-sensitivity of this

https://github.com/moby/moby/blob/d3579b9582d44df798fd8a85f39193a569402ee1/daemon/graphdriver/overlay2/overlay.go#L231-L251